src/modalities/dataloader/create_index.py

```diff
@@ -59,16 +60,16 @@ def queue_generator():
         def process_line(last_index: int, curr_index: int):
             segment_len = curr_index - last_index
             try:  # check if line is a valid json
-                line = np.memmap(self.src_file, mode="r", offset=last_index, shape=(segment_len,)).view("S1").tolist()
-                line = [c.decode("utf8") for c in line]
-                line = "".join(line)
-                json.loads(line)
+                f = self.src_file.open(encoding=DEFAULT_ENCODING)
```

Comment:
We can do this, however if we really want to improve speed here, parallelizing the processing and writing is the way to go. In this case, this file descriptor must stay within this function to avoid concurrency bugs. 
I will add this usage of a shared file descriptor now, but if we extend this in the future, we need to keep this in mind!

Reply:
I redid the benchmarks. Unfortunately there was no significant improvement. Maybe this might get visible when using really large datasets. But as I mentioned before, if we really wanna boost this, another parallelization effort is the way to go. Meaning using multiprocesses for the process_line calls and threading for the writer_thread. Both require some work to avoid race conditions, deadlocks and proper workload distribution